### DISCUSSION TITLE
Cannot add new router on new version Router OS

### DISCUSSION THREAD

**Discussion by axmad386** [hotspotbilling/phpnuxbill#97] [Q&A]
**Describe the bug**
Cannot add Router with valid data on Newer version of RouterOS

**To Reproduce**
1. Go to 'Network->Router->New Router'
2. Fill valid router data
3. Save
4. Error will shown
```
Invalid username or password supplied.

#0 /home/dmuggle/FL/phpnuxbill/system/autoload/Mikrotik.php(24): PEAR2\Net\RouterOS\Client->__construct()
#1 /home/dmuggle/FL/phpnuxbill/system/controllers/routers.php(159): Mikrotik::getClient()
#2 /home/dmuggle/FL/phpnuxbill/system/boot.php(337): include('...')
#3 /home/dmuggle/FL/phpnuxbill/index.php(22): require_once('...')
#4 {main}
```

**Expected behavior**
New Router is saved

**Additional context**
Router OS version:  v6.42.12.

**Workaround**
Just update Client.php to dev branch
https://github.com/pear2/Net_RouterOS/blob/develop/src/PEAR2/Net/RouterOS/Client.php
See this related issue https://github.com/pear2/Net_RouterOS/issues/37

**Comment by ibnux** [comment]
Everything working fine

**Comment by axmad386** [comment]
Are you sure sir? Which RouterOS version you are using? Because in my case I cannot log in without using pear/Net_RouterOS dev version

**Comment by ibnux** [comment]
if it error, all user will report to me
and my billing still working, you have bad configuration

https://t.me/phpnuxbill
